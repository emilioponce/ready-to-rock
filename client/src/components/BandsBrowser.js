import React, { Component } from 'react'
import BandsList from './BandsList'
import SearchBox from './SearchBox'
import { getBandsIds } from '../api'

import styles from './BandsBrowser.module.css'

class BandsBrowser extends Component {
  _isMounted = false

  constructor(props) {
    super(props)
    this.state = {
      bandsIds: null
    }
  }

  async componentDidMount() {
    this._isMounted = true
    // init with all the band
    const bandsIds = await getBandsIds()
    if (this._isMounted) {
      bandsIds
        ? this.setState({ bandsIds: bandsIds.data })
        : console.log('Error fetching Bands ID list')
    }
  }

  componentWillUnmount() {
    this._isMounted = false
  }

  handleSearchTermChange = bandsIds => {
    this.setState({ bandsIds: bandsIds.data })
  }

  reloadBrowser = () => {
    window.location.reload()
  }

  render() {
    const { bandsIds } = this.state

    // conditional renderings
    if (bandsIds === null) {
      return <div className={styles.content}>Loading ...</div>
    }
    if (bandsIds.length === 0) {
      return (
        <div className={styles.content}>
          Bands not found.{' '}
          <button className={styles.button} onClick={this.reloadBrowser}>
            Return
          </button>
        </div>
      )
    }

    return (
      <div className={styles.content}>
        <SearchBox onSearchTermChange={this.handleSearchTermChange} />
        <BandsList bandsIds={bandsIds} />
      </div>
    )
  }
}

export default BandsBrowser
