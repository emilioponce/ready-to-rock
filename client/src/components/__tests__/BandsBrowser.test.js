import React from 'react'
import ReactDOM from 'react-dom'
import BandsBrowser from '../BandsBrowser'
import { shallow } from 'enzyme'

const component = <BandsBrowser />

jest.mock('../BandsList')

describe('Component BandsBrowser', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('snapshot renders correctly', () => {
    const wrapper = shallow(component)
    expect(wrapper).toMatchSnapshot()
  })

  it("when bands list is null 'Loading...' message renders", () => {
    const wrapper = shallow(component)
    const text = wrapper.text()
    expect(text).toBe('Loading ...')
  })

  it("when bands length is 0 'Bands not found' message renders", () => {
    const wrapper = shallow(component)
    wrapper.setState({ bandsIds: [] })
    const text = wrapper.text()
    expect(text).toBe('Bands not found. Return')
  })
})
