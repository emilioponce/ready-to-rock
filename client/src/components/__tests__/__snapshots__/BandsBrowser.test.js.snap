// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component BandsBrowser snapshot renders correctly 1`] = `
<div
  className="content"
>
  Loading ...
</div>
`;
